feat: Implement base code for a c8 license check

matthewBearCamunda opened the pull request:
## Description

Adds a new module, `camunda-commons`, which is intended to be available to be used by all apps in the monorepo. This new module has a sub-module `camunda-license` which is is used to read and determine if license is valid or not. The license is set using an environment variable

This PR is not the full implementation, but rather setting a foundation for future changes. The file structure, tests, and basic code skeleton for C8 Licenses is included in this PR

## Related issues

closes https://github.com/camunda/camunda/issues/18969

romansmirnov reviewed (commented):
@matthewBearCamunda, thanks for creating the PR.

I went through it, technically, it works but I was giving it a second thought, I strongly believe we should not introduce yet another way on how to implement services and apis (like the license check and others). Instead, we should build this upon the - just today - [merged service architecture](https://github.com/camunda/camunda/pull/19049).

Basically, the service infrastructure could be extended with another service like `ManagementService` with, for example, such an API:
```
CamundaServices services = ...;
final var result = services.managementService().validateLicense();
result.isValid(); // and so on
```
This new service can then be used in the REST layer by implementing a dedicated REST controller that calls that service. For further details, this slide deck should give more context: https://docs.google.com/presentation/d/1dnnXgx7UENeyf1wYGiYPq7-YVnlTTp8814Q6veZj9P0/edit - I think it is important that we follow that path, and not that each introduces their own way.

Other thoughts:
1. I don't understand why Spring is required to implement a "cache"?
  a. In our current architecture, the usage of Spring is as much as possible minized just to our needs. For example, we use it to configure the application and declare a minimum of beans. Everything else "pure" Java implementation. That allows us to write "simple" unit tests without external dependencies like Spring, etc.
  b. Basically, I don't want to say that using Spring is something bad, but if it does not solve a specific problem, then I do recommend not using (and with a specific problem I don't mean autowiring beans). For example, we are using Spring to implement the REST controller.
2. The PR introduces new test dependencies that are not part of the stack so far. As above, the same applies, we should follow existing patterns to write test cases. It makes it easier for us engineers to write test cases, and we don't need to understand depending on the current service/context how to write test cases. Basically, we should first see if the existing test dependencies and the way how to write test cases are sufficient. If not, then let's consider alternatives.

However, the most important part we have to agree on that we use the service architecture to implement such a license check as a service, that can be used by REST or any other service when needed.

Do you want to sync and discuss how a potential implementation could look like (with the new service architecture)? Feel free to schedule a meeting.

matthewBearCamunda commented:
@romansmirnov - Thanks for the review

I refactored the implementation to use the new service architecture. I also removed the test dependency in favor of mocking the environment variable read 

It's ready for a re-review!

romansmirnov reviewed (approved):
@matthewBearCamunda, that looks simple and easy, thanks for applying my feedback. I like that direction, LGTM!
